File: RemixedDungeon/src/main/java/com/watabou/pixeldungeon/actors/buffs/Cripple.java

package com.watabou.pixeldungeon.actors.buffs;

import com.nyrds.pixeldungeon.ml.R;
import com.nyrds.platform.util.StringsManager;
import com.watabou.pixeldungeon.sprites.CharSprite;
import com.watabou.pixeldungeon.ui.BuffIndicator;

public class Cripple extends FlavourBuff {

	public static final float DURATION	= 10f;
	
	@Override
	public int icon() {
		return BuffIndicator.CRIPPLE;
	}

	@Override
	public float speedMultiplier() {
		return 0.5f;
	}

	@Override
	public void attachVisual() {
        target.showStatus(CharSprite.NEGATIVE, StringsManager.getVar(R.string.Char_StaCrippled));
	}
}

File: scripts/buffs/Cripple.lua
---
--- Cripple: halves movement speed (was actors/buffs/Cripple.java)
---
local RPD  = require "scripts/lib/commonClasses"
local buff = require "scripts/lib/buff"

local CharSprite = luajava.bindClass("com.watabou.pixeldungeon.sprites.CharSprite")

return buff.init{
    desc  = function ()
        return {
            icon          = 23, -- BuffIndicator.CRIPPLE
            name          = "CrippleBuff_Name",
            info          = "CrippleBuff_Info",
        }
    end,

    speedMultiplier = function(self, buff)
        return 0.5
    end,

    attachVisual = function(self, buff)
        buff.target:showStatus(CharSprite.NEGATIVE, RPD.textById("Char_StaCrippled"))
    end
}
